### ISSUE TITLE
`fisher` fails to function if no packages are installed on V3

### ISSUE THREAD

**Issue by onbjerg** [jorgebucaran/fisher#455]
Fresh install of `fisher` yields an error on [this line](https://github.com/jorgebucaran/fisher/blob/master/fisher.fish#L147) due to there being no packages, so the expansion fails.

Reproduce:
1. Install `fisher` on a system that has never had `fisher`
2. Try to run `fisher` or `fisher add`

**Log**

```
~/.config/fish/functions/fisher.fish (line 1): No matches for wildcard '$fisher_config/*/*/*'. See `help expand`.
_fisher_pkg_remove_all $fisher_config/*/*/*
                       ^
in command substitution
    called on line 144 of file ~/.config/fish/functions/fisher.fish

in function '_fisher_commit'
    called on line 59 of file ~/.config/fish/functions/fisher.fish

in function 'fisher'
    called on standard input

nothing to commit -- try adding some packages
```

**Comment by jorgebucaran** [comment]
@onbjerg Could you please post exactly the commands you typed and entered on the command line? Thanks.

**Comment by onbjerg** [comment]
@jorgebucaran 

Above error is from just running `fisher`.

Running `fisher add oh-my-fish/theme-robbyrussell` yields this error:

```
~/.config/fish/functions/fisher.fish (line 1): No matches for wildcard '$fisher_config/*/*/*'. See `help expand`.
_fisher_pkg_remove_all $fisher_config/*/*/*
                       ^
in command substitution
    called on line 144 of file ~/.config/fish/functions/fisher.fish

in function '_fisher_commit'
    called on line 59 of file ~/.config/fish/functions/fisher.fish
    with parameter list 'add oh-my-fish/theme-robbyrussell'

in function 'fisher'
    called on standard input
    with parameter list 'add oh-my-fish/theme-robbyrussell'

fetching https://codeload.github.com/oh-my-fish/theme-robbyrussell/tar.gz/master
linking ~/.config/fish/functions/fish_prompt.fish
added 1 package in 0.75s
```

It installs the package successfully though, and then all errors going forward (so far) disappear altogether.

**Comment by jorgebucaran** [comment]
@onbjerg Could you tell me your fish version and operating system?

I'm unable to reproduce this on a fresh fisher install.

```console
x@oni ~> ll ~/.config/fish/completions/ ~/.config/fish/functions/
/Users/x/.config/fish/completions/:

/Users/x/.config/fish/functions/:
total 8
-rw-r--r--  1 x  staff   102B Sep 29 21:53 fish_user_key_bindings.fish
x@oni ~> fisher
fish: Unknown command 'fisher'
x@oni ~> curl https://git.io/fisher --create-dirs -sLo ~/.config/fish/functions/fisher.fish
x@oni ~> fisher
created empty fishfile in ~/.config/fish/fishfile
nothing to commit -- try adding some packages
```

**Comment by jorgebucaran** [comment]
@onbjerg Nevermind, I was using fish 2.0 (for testing purposes) and failed to remember fish 2.3+ changed the default behavior. I can reproduce this now. This is a bug.

- https://github.com/fish-shell/fish-shell/pull/2719
- https://github.com/fish-shell/fish-shell/issues/4175

cc @faho

**Comment by onbjerg** [comment]
It is on a colleagues system, he just installed `fish` and I recommended him to install `fisher`, unaware that V3 was released. So can confirm he is running `fish` 2.3+

**Comment by jorgebucaran** [comment]
@onbjerg Should be fixed in 3.0.3 (444a8fee8cbdeaf58ad3dcdb537a18037116c8ad). 🎉 

Please use `fisher self-update` to pull the changes.

**Comment by jorgebucaran** [comment]
@onbjerg Should be fixed in 3.0.3 (`13eba77`). 🎉 

Please use `fisher self-update` to pull the changes.
